src/kubernetes/namespaces.go

```diff
@@ -0,0 +1,35 @@
+package kubernetes
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// GetNamespace will get the namespace object by the namespace name
+func (kc *KubeCtl) GetNamespace(name string) (*corev1.Namespace, error) {
+	return kc.Clientset.CoreV1().Namespaces().Get(name, metav1.GetOptions{})
+}
+
+// GetNamespaces will get all namespaces from the k8s cluster
+func (kc *KubeCtl) GetNamespaces() ([]*corev1.Namespace, error) {
+	namespaces := []*corev1.Namespace{}
+	namespacesList, err := kc.Clientset.CoreV1().Namespaces().List(metav1.ListOptions{})
+	if err != nil {
+		return namespaces, err
+	}
+	for _, n := range namespacesList.Items {
+		namespaces = append(namespaces, &n)
+	}
+	return namespaces, nil
+}
+
+// CreateNamespace will create the namespace by the namespace object
+func (kc *KubeCtl) CreateNamespace(namespace *corev1.Namespace) (*corev1.Namespace, error) {
+	return kc.Clientset.CoreV1().Namespaces().Create(namespace)
+}
+
+// DeleteNamespace will delete the namespace by the namespace name
+func (kc *KubeCtl) DeleteNamespace(name string) error {
+	deletePolicy := metav1.DeletePropagationBackground
```

Comment:
Do we need DeletePropagationBackground option here?

Reply:
I referred to https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/namespace/deletion/namespaced_resources_deleter.go#L340-L344
IMO, maybe it's good to need?
What do you think?

Reply:
I think we should use **Foreground** waiting for all resource to be killed. see https://godoc.org/k8s.io/apimachinery/pkg/apis/meta/v1#DeletionPropagation

```
    // The object exists in the key-value store until the garbage collector
    // deletes all the dependents whose ownerReference.blockOwnerDeletion=true
    // from the key-value store.  API sever will put the "foregroundDeletion"
    // finalizer on the object, and sets its deletionTimestamp.  This policy is
    // cascading, i.e., the dependents will be deleted with Foreground.
```